latency.go

```diff
@@ -0,0 +1,87 @@
+package l2
+
+import (
+	"log"
+	"time"
+)
+
+// A special struct that contains an embedded tap device. It's meant to be
+// used as a drop-in replacement for a normal tapDevice. It's main power is
+// adding latency to packet reads and writes.
+type tapDeviceWithLatency struct {
```

Comment:
I'd just make something which takes a FrameReader and controls latency, similar to Logger